src/cli.rs

```diff
@@ -1135,6 +1143,20 @@ impl Opt {
         git_config: Option<GitConfig>,
         assets: HighlightingAssets,
     ) -> Self {
+        let matches = Self::command().get_matches();
+
+        if let Some(path) = matches.get_one::<String>("config") {
+            if !path.is_empty() {
+                let parsed_config = GitConfig::try_create_from_path(&env, path);
+                return Self::from_clap_and_git_config(
+                    env,
+                    Self::command().get_matches(),
+                    parsed_config,
+                    assets,
+                );
+            }
+        }
+
         Self::from_clap_and_git_config(env, Self::command().get_matches(), git_config, assets)
```

Comment:
Nit: if you can find a nice way of doing it I'd prefer to avoid duplicating the

```
Self::from_clap_and_git_config(env, Self::command().get_matches(), git_config, assets)
```

call and instead compute `git_config` either from the custom file or accepting the passed-in value.

Reply:
Done. What about now?